spec.html

```diff
@@ -32118,11 +32204,25 @@ <h1>
         </h1>
         <dl class="header">
           <dt>description</dt>
-          <dd>It converts _t_ from UTC to local time.</dd>
+          <dd>
+            It converts _t_ from UTC to local time.
+            The local political rules for standard time and daylight saving time in effect at _t_ should be used to determine the result in the way specified in this section.
+          </dd>
         </dl>
         <emu-alg>
-          1. Return _t_ + LocalTZA(_t_, *true*).
+          1. Let _localTimeZone_ be DefaultTimeZone().
+          1. Let _offsetParseResult_ be Completion(ParseTimeZoneOffsetString(_localTimeZone_)).
+          1. If _offsetParseResult_ is a normal completion, then
+            1. Let _offsetNs_ be _offsetParseResult_.[[Value]].
+          1. Else,
+            1. Let _offsetNs_ be GetNamedTimeZoneOffsetNanoseconds(_localTimeZone_, ℤ(ℝ(_t_) &times; 10<sup>6</sup>)).
```

Comment:
I don't understand why we use this fallback pattern instead of just

```suggestion
          1. Let _offsetNs_ be GetNamedTimeZoneOffsetNanoseconds(_localTimeZone_, ℤ(ℝ(_t_) &times; 10<sup>6</sup>)).
```

Also, if this pattern is necessary, could we encapsulate it into its own AO? I think the swallowing of a completion record here is a bit strange and I'd rather limit it to one occurrence.

Reply:
It's because unlike the offset nanoseconds of a named time zone like `America/Sao_Paulo`, it's fully specifiable how to get the offset nanoseconds of a time zone like `-07:00` or `+05:30` or `+00:19:32.13`.

The weirdness with the completion record is because we don't have any other mechanism than ParseText for determining whether a string matches a grammar. (Previously, in Temporal, I had language like "`if _string_ satisfies the syntax of |GrammarSymbol|`" but we considered that not rigorous enough.)

Here's a proposal that I think might address your concerns:

1. If IsTimeZoneOffsetString(_localTimeZone_) is **true**, then
  - a. Let _offsetNs_ be ! ParseTimeZoneOffsetString(_localTimeZone_).
2. Else,
  - a. Let _offsetNs_ be GetNamedTimeZoneOffsetNanoseconds(_localTimeZone_, ...).

IsTimeZoneOffsetString would be a new AO that encapsulates using ParseText to determine whether the string is an offset string. That way, we don't have to use completion records at all.

Reply:
That WFM. I would've probably gone a step further and encapsulated the rest of that conditional, but either way is fine.